core/lib/libtomcrypt/src/tee_ltc_provider.c

```diff
@@ -1203,13 +1206,19 @@ static TEE_Result rsassa_verify(uint32_t algo, struct rsa_public_key *key,
 		goto err;
 	}
 
+	if (ltc_rsa_algo == LTC_PKCS_1_PSS && salt_len == -1)
+		salt_len = hash_size;
+
 	ltc_res = rsa_verify_hash_ex(sig, sig_len, msg, msg_len, ltc_rsa_algo,
 				     ltc_hashindex, salt_len, &stat, &ltc_key);
-	if ((ltc_res != CRYPT_OK) || (stat != 1)) {
-		res = TEE_ERROR_SIGNATURE_INVALID;
-		goto err;
-	}
-	res = TEE_SUCCESS;
+
+	if (ltc_res == CRYPT_OK)
+		if (ltc_stat == 1)
+			res = TEE_SUCCESS;
+		else
+			res = TEE_ERROR_SIGNATURE_INVALID;
+	else
+		res = TEE_ERROR_GENERIC;
```

Comment:
Well it looks like this breaks xtest 4011 (other that that, all regression tests are OK):

```
* regression_4011 Test TEE Internal API Bleichenbacher attack (negative)
Iteration 0
ERROR:   USER-TA:TEE_AsymmetricVerifyDigest:1756: Panic 0xffff0000
ERROR:   [0x0] TEE-CORE:tee_svc_sys_return_helper:256: TA panicked with code 0xffff0000 usr_sp 0x80000f80 usr_lr 0x5046d720
/home/jerome/work/d02/d02_optee/optee_test/host/xtest/regression_4000.c:779: ret_orig has an unexpected value: 0x3 = TEEC_ORIGIN_TEE, expected 0x4 = TEEC_ORIGIN_TRUSTED_APP
/home/jerome/work/d02/d02_optee/optee_test/host/xtest/regression_4000.c:5120: ta_crypt_cmd_asymmetric_verify(c, &s, opv, ((void *)0), 0, in, sizeof(in), out, out_size) has an unexpected value: 0xffff3024 = TEE_ERROR_TARGET_DEAD, expected 0xffff3072 = TEE_ERROR_SIGNATURE_INVALID
  regression_4011 FAILED
```

But I would think the test is incorrect. IMO it should check `status != TEE_SUCCESS` rather than `status == TEE_ERROR_SIGNATURE_INVALID`. @jenswi-linaro what do you think?